calico_containers/adapter/datastore.py

```diff
@@ -485,16 +580,14 @@ def add_bgp_peer(self, version, ip):
         """
         assert version in ("v4", "v6")
         assert isinstance(ip, IPAddress)
-        bgp_peer_path = BGP_PEER_PATH % {"version": version}
-
-        # Check if the peer exists.
-        if ip in self.get_bgp_peers(version):
-            return
-
-        self.etcd_client.write(bgp_peer_path, str(ip), append=True)
+        bgp_peer_path = HOST_BGP_PEER_PATH % {"hostname": hostname,
```

Comment:
This subtly changes the behavior of `add_node_bgp_peer`.  Before it did nothing if the peer existed.  Now it will overwrite the AS number of the peer.  Is that what we want?  Or should we not overwrite and throw an error?

I don't much care, but you need to update the comment to say the right thing!